src/Microsoft.Diagnostics.ExtensionCommands/FindEphemeralReferencesToLOHCommand.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Diagnostics.DebugServices;
+using Microsoft.Diagnostics.Runtime;
+using static Microsoft.Diagnostics.ExtensionCommands.TableOutput;
+
+namespace Microsoft.Diagnostics.ExtensionCommands
+{
+    [Command(Name = "FindEphemeralReferencesToLOH", Help = "Finds ephemeral objects which reference the large object heap.")]
+    public class FindEphemeralReferencesToLOHCommand : CommandBase
+    {
+        // IComparer for binary search
+        private readonly IComparer<(ClrObject, ClrObject)> _firstObjectComparer = Comparer<(ClrObject, ClrObject)>.Create((x, y) => x.Item1.Address.CompareTo(y.Item1.Address));
+
+        [ServiceImport]
+        public ClrRuntime Runtime { get; set; }
+
+        public override void Invoke()
+        {
+            int segments = Runtime.Heap.Segments.Count(seg => seg.Kind is not GCSegmentKind.Frozen or GCSegmentKind.Pinned);
+            if (segments > 50)
+            {
+                string gcSegKind = Runtime.Heap.SubHeaps[0].HasRegions ? "regions" : "segments";
+                Console.WriteLineWarning($"Walking {segments:n0} {gcSegKind}, this may take a moment...");
+            }
+
+            TableOutput output = new(Console, (16, "x12"), (64, ""), (16, "x12"));
+
+            // Ephemeral -> Large
+            List<(ClrObject From, ClrObject To)> ephToLoh = FindEphemeralToLOH().OrderBy(i => i.From.Address).ThenBy(i => i.To.Address).ToList();
+            if (ephToLoh.Count == 0)
+            {
+                Console.WriteLine("No Ephemeral objects pointing to the Large objects.");
```

Comment:
CommandBase does have WriteLine impls that adds a CTRL-C check ( Console.CancellationToken.ThrowIfCancellationRequested()) for commands with a lot of output or in a loop.   It is still ok to do Console.WriteLine but you lose the automatic CTRL-C.  I'm not the CommandBase.WriteLine is the right way to automatically add the CTRL-C check, but I really don't think they should be in the console service impl itself or in other services.  I think the checks should be in the commands themselves hence the CommandBase.WriteLine. 

In the end, you can still use Console.WriteLine everywhere but you may want to add the above cancelation call to any long loops.